src/bootlint.js

```diff
@@ -1100,6 +1104,43 @@ var LocationIndex = _location.LocationIndex;
             reporter('`.modal-dialog` must have a `role="document"` attribute.', modalDialogs);
         }
     });
+    addLinter("E050", function requireFormControlClass($, reporter) {
+        //Select inputs of the proper type that do not have a `.form-control` class OR a `.col-*-*` class
+        var colSelector = COL_CLASSES.join(",");
+        var inputs = $('select:not(.form-control,' + colSelector + '), textarea:not(.form-control,' + colSelector + '), ' + (FORM_CONTROL_VALID_TYPES.map(function (type) {
+                return 'input:not(.form-control,' + colSelector + ')[type="' + type + '"]';
+            }).join(',')
+        ));
+        if (inputs.length) {
+            inputs.each(function (i, el) {
+                reporter('The `.form-control` class must appear on all textual `<input/>` elements, `<textarea>` elements, and `<select>` elements', $(el));
+            });
+        }
+    });
+    addLinter("E051", function requireCheckboxClass($, reporter) {
+        //Checkboxes are allowed to exist inside of `label.btn` when used with the buttons.js plugin: http://getbootstrap.com/javascript/#buttons-checkbox-radio
+        //Checkboxes are allowed to exist inside of input groups as addons: http://getbootstrap.com/components/#input-groups-checkboxes-radios
+        //We will ignore checkboxes that incorrectly have the `.form-control` class since rule E042 handles warning about this
```

Comment:
Alright, that makes it simpler.  I'll remove this check, but for now E042 will fail the tests until these rules are disabled by default.